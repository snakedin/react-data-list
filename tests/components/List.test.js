import React from "react";
import { render } from "@testing-library/react";
import DataList, { Column } from "../../src/index";

it("DataList render", () => {

    const provider = [];
    let result;

    result = render(
        <DataList/>
    );
    expect(result.asFragment()).toMatchSnapshot('#1 error on component without any props');

    result = render(
        <DataList provider={ provider }/>
    );
    expect(result.asFragment()).toMatchSnapshot('#2 error on empty columns');

    result = render(
        <DataList>
            <Column id="id"/>
        </DataList>
    );
    expect(result.asFragment()).toMatchSnapshot('#3 error on empty provider');

    result = render(
        <DataList provider={ provider }>
            <Column id="id" />
        </DataList>
    );
    expect(result.asFragment()).toMatchSnapshot('#4 all default props');

    result = render(
        <DataList provider={ provider } enableSorting={ false } showFilter={ false }>
            <Column id="id" />
        </DataList>
    );
    expect(result.asFragment()).toMatchSnapshot('#5 without sorting and filtering');
});
