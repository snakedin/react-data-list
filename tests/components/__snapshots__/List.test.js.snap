// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataList render: #1 error on component without any props 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Error 
    </h1>
    <p>
      [Error]: Unknown provider type
    </p>
  </div>
</DocumentFragment>
`;

exports[`DataList render: #2 error on empty columns 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Error 
    </h1>
    <p>
      At least one column is required
    </p>
  </div>
</DocumentFragment>
`;

exports[`DataList render: #3 error on empty provider 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Error 
    </h1>
    <p>
      [Error]: Unknown provider type
    </p>
  </div>
</DocumentFragment>
`;

exports[`DataList render: #4 all default props 1`] = `
<DocumentFragment>
  <div
    class="data-list"
  >
    <div>
      <table
        class="table table-striped table-hover table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              <span
                class="btn btn-sm btn-link asc"
              >
                Id
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <input
                class="form-control"
                type="text"
                value=""
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DataList render: #5 without sorting and filtering 1`] = `
<DocumentFragment>
  <div
    class="data-list"
  >
    <div>
      <table
        class="table table-striped table-hover table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              <span
                class="btn btn-sm"
              >
                Id
              </span>
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
